tests/platform_tests/link_flap/conftest.py

```diff
@@ -26,23 +25,20 @@ def pytest_addoption(parser):
 
 
 @pytest.fixture()
-def bring_up_fanout_interfaces(request, all_ports, duthosts, fanouthosts):
+def bring_up_fanout_interfaces(request, duthosts, fanouthosts):
     """
     Bring up outer interfaces on the DUT.
 
     Args:
         request: pytest request object
-        duthost: Fixture for interacting with the DUT.
+        duthosts: Fixture for interacting with the DUT list.
         fanouthosts: Fixture for interacting with the fanouts.
     """
     yield
     if request.node.rep_call.failed:
-        dutname, portname = decode_dut_port_name(all_ports)
-
         for dut in duthosts:
-            if dutname == 'unknown' or dutname == dut.hostname:
-                candidates = build_test_candidates(dut, fanouthosts, portname)
-                for _, fanout, fanout_port in candidates:
-                    fanout.no_shutdown(fanout_port)
+            candidates = build_test_candidates(dut, fanouthosts, "all_ports")
+            for _, fanout, fanout_port in candidates:
+                fanout.no_shutdown(fanout_port)
```

Comment:
I wonder if, just to be safe, we shouldn't condition this on `if request...failed`? Would it hurt to just bring up all the fanout ports at the end every time? (this doesn't need to be changed in this PR)

Reply:
Also, can we swap that `sleep` for a `wait_until` where we check if all the active DUT ports are showing as oper up? (this can also be improved upon later)

Reply:
Yeah. This is a stop bleeding change. There is more room for improvement.